src/test/testUtil.ts

```diff
@@ -146,46 +146,69 @@ export function getMetrics<K extends MetricName>(
-    eventNum?: number
+    expected: MetricShapes[K] | MetricShapes[K][]
 ): void | never
 export function assertTelemetry<K extends MetricName>(
     name: K,
-    expected: MetricShapes[MetricName],
-    eventNum?: number
+    expected: MetricShapes[MetricName] | MetricShapes[MetricName][]
 ): void | never
 export function assertTelemetry<K extends MetricName>(
     name: K,
-    expected: MetricShapes[K],
-    eventNum: number = 0
+    expected: MetricShapes[K] | MetricShapes[K][]
 ): void | never {
-    const expectedCopy = { ...expected } as { -readonly [P in keyof MetricShapes[K]]: MetricShapes[K][P] }
-    const passive = expectedCopy?.passive
+    const expectedList = Array.isArray(expected) ? expected : [expected]
     const query = { metricName: name, excludeKeys: ['awsAccount', 'duration'] }
-    delete expectedCopy['passive']
-
-    Object.keys(expectedCopy).forEach(
-        k => ((expectedCopy as any)[k] = (expectedCopy as Record<string, any>)[k]?.toString())
-    )
-
-    // Telemetry client should add awsRegion to all metrics.
-    ;(expectedCopy as any)['awsRegion'] = globals.regionProvider.guessDefaultRegion()
-
-    const metadata = globals.telemetry.logger.query(query)
-    assert.ok(metadata.length > 0, `Telemetry did not contain any metrics with the name "${name}"`)
-    assert.deepStrictEqual(metadata[eventNum], expectedCopy)
+    let metadata = globals.telemetry.logger.query(query)
+    assert.ok(metadata.length > 0, `telemetry not found for metric name: "${name}"`)
+
+    for (let i = 0; i <= expectedList.length; i++) {
+        const metric = expectedList[i]
+        const expectedCopy = { ...metric } as { -readonly [P in keyof MetricShapes[K]]: MetricShapes[K][P] }
+        const passive = expectedCopy?.passive
+        delete expectedCopy['passive']
+
+        Object.keys(expectedCopy).forEach(
+            k => ((expectedCopy as any)[k] = (expectedCopy as Record<string, any>)[k]?.toString())
+        )
+
+        // Telemetry client adds awsRegion to all metrics.
+        ;(expectedCopy as any)['awsRegion'] = globals.regionProvider.guessDefaultRegion()
+
+        if (expectedList.length == 1) {
+            assert.deepStrictEqual(metadata[0], expectedCopy)
+        } else {
+            const found = metadata.findIndex(val => {
+                // Hacky way to "deep compare".
+                try {
+                    assert.deepStrictEqual(val, expectedCopy)
+                } catch (e) {
+                    return false
+                }
+                return true
+            })
+            assert.ok(
+                found >= 0,
+                `telemetry item ${i + 1} (of ${
+                    expectedList.length
+                }) not found (in the expected order) for metric name: "${name}" `
+            )
+            // Remove all items up to the found item, to ensure we are checking the expected _order_.
+            metadata = metadata.splice(found, metadata.length - 1)
```

Comment:
Checking "non-continguous order" is something generalizable (maybe already is). This is a hacky way to do it for now, but at least the interface is here for use in tests.

Reply:
`metadata` is copied though, so no harm in mutating it? Will revisit later, want to unblock this PR for now.